pkg/volume/glusterfs/glusterfs.go

```diff
@@ -141,12 +141,49 @@ func (plugin *glusterfsPlugin) NewMounter(spec *volume.Spec, pod *v1.Pod, _ volu
 	if kubeClient == nil {
 		return nil, fmt.Errorf("glusterfs: failed to get kube client to initialize mounter")
 	}
+
 	ep, err := kubeClient.Core().Endpoints(ns).Get(epName, metav1.GetOptions{})
 	if err != nil {
 		glog.Errorf("glusterfs: failed to get endpoints %s[%v]", epName, err)
-		return nil, err
+
+		class, err := volutil.GetClassForVolume(plugin.host.GetKubeClient(), spec.PersistentVolume)
```

Comment:
recreate endpoint if:
- spec != nil
- err indicates the endpoint doesn't exist.